# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/style/FontFaceSetDocumentImpl.cpp
# Commit: 14036a42457b
# Full Hash: 14036a42457bdcc038fd3f88b6e77523a60ecc9b
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2022-07-12 04:20:08
# Description:
#   Bug 1779098 - Ensure we check FontFaceSetDocumentImpl::mDocument for nullptr. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D151557
# ==============================================================================

diff -r 81cd4125a2d2 -r 14036a42457b layout/style/FontFaceSetDocumentImpl.cpp
--- a/layout/style/FontFaceSetDocumentImpl.cpp	Mon Jul 11 23:24:36 2022 +0000
+++ b/layout/style/FontFaceSetDocumentImpl.cpp	Mon Jul 11 23:24:47 2022 +0000
@@ -150,6 +150,9 @@
 void FontFaceSetDocumentImpl::RefreshStandardFontLoadPrincipal() {
   MOZ_ASSERT(NS_IsMainThread());
   RecursiveMutexAutoLock lock(mMutex);
+  if (NS_WARN_IF(!mDocument)) {
+    return;
+  }
   mStandardFontLoadPrincipal = MakeRefPtr<gfxFontSrcPrincipal>(
       mDocument->NodePrincipal(), mDocument->PartitionedPrincipal());
   FontFaceSetImpl::RefreshStandardFontLoadPrincipal();
@@ -213,6 +216,10 @@
 #endif
 
 bool FontFaceSetDocumentImpl::Add(FontFaceImpl* aFontFace, ErrorResult& aRv) {
+  if (NS_WARN_IF(!mDocument)) {
+    return false;
+  }
+
   if (!FontFaceSetImpl::Add(aFontFace, aRv)) {
     return false;
   }
@@ -233,6 +240,10 @@
 
 nsresult FontFaceSetDocumentImpl::StartLoad(gfxUserFontEntry* aUserFontEntry,
                                             uint32_t aSrcIndex) {
+  if (NS_WARN_IF(!mDocument)) {
+    return NS_ERROR_FAILURE;
+  }
+
   nsresult rv;
 
   nsCOMPtr<nsIStreamLoader> streamLoader;
@@ -326,6 +337,10 @@
     return true;
   }
 
+  if (NS_WARN_IF(!mDocument)) {
+    return false;
+  }
+
   RefPtr<gfxFontSrcPrincipal> gfxPrincipal =
       aSrc.mURI->InheritsSecurityContext() ? nullptr
                                            : aSrc.LoadPrincipal(*this);
